src/components/TableObjectBody/TableObjectBody.tsx

```diff
@@ -1,148 +0,0 @@
-import clsx from 'clsx';
-import React, { ReactNode, useEffect, useState, useRef, UIEvent } from 'react';
-import styles from '../TableObject/TableObject.module.css';
-
-export interface Props {
-  /**
-   * Behavior variations:
-   * - **overflow** renders a table object body that overflows horizontally off the screen
-   */
-  behavior?: 'overflow';
-  /**
-   * Child node(s) that can be nested inside component
-   */
-  children: ReactNode;
-  /**
-   * CSS class names that can be appended to the component.
-   */
-  className?: string;
-}
-
-/**
- * BETA: This component is still a work in progress and is subject to change.
- *
- * ```ts
- * import {TableObjectBody} from "@chanzuckerberg/eds";
- * ```
- *
- * Component TableObjectBody is used in the TableObject component to display the content with shadow gradient and overflow scroll behaviour.
- */
-export const TableObjectBody = ({
-  behavior,
-  children,
-  className,
-  ...other
-}: Props) => {
-  /**
-   * Set states and refs
-   *
-   * Set isEnd state: set to true, right shadow gradient activates. Removed when false.
-   */
-  const [isEnd, setIsEnd] = useState(true);
-  /**
-   * Set isState state: set to true, left shadow gradient activates. Removed when false.
-   */
-  const [isStart, setIsStart] = useState(false);
-  /**
-   * Target table body and table body inner DOM elements.
-   */
-  const tableObjectBodyRef = useRef<HTMLDivElement>(null);
-  const tableObjectBodyInnerRef = useRef<HTMLDivElement>(null);
-
-  /**
-   * Set right and left gradients on tables
-   */
-  const setShadows = () => {
```

Comment:
Are we thinking that tables won't use this behavior or they'll implement it themselves every time? If it's the latter, should we move this logic into a story so devs can see how it's done?

Reply:
This adds masking like in `Tabs` to indicate overflow.

However I don't think we should own overflow behavior and the parent can add as wished either via `overflow: scroll` to always show scroll bar or some masking.

Is there a way to extract the masking from Tabs and export it for use? Kind of like our mixins but also for available outside EDS

Reply:
> Is there a way to extract the masking from Tabs and export it for use? Kind of like our mixins but also for available outside EDS

I haven't thought about it, but it would be nice to make that reusable if it's a pattern we want to keep using. I remember seeing it on a table design that was brought to office hours once, where they had one column fixed in view and the rest of the table scrolled, and I think they were using some sort of fade to indicate a section of the row was scrolled out of view.

Reply:
If you don't mind, I'm going to punt adding the overflow story and the masking into another ticket

Reply:
Sounds good 👍

Reply:
As an update to this, I had a discussion with @ahuth and agree that we should keep the table component dumber and not display overflow behavior, as:
- discourage overpopulating tables
- discourage use in Modals
- would require a div wrapper with use case specific functionality
- should leave it up to consumers to handle their own use case (and hence discouraging cases where tables need overflow and is not resolved by whole page scrolling)